api/src/test/java/org/openmrs/module/commonreports/reports/LabReportManagerTest.java

```diff
@@ -0,0 +1,170 @@
+import java.math.BigDecimal;
+import static org.junit.Assert.assertEquals;
+
+public class LabReportManagerTest extends BaseModuleContextSensitiveMysqlBackedTest {
+	
+	public LabReportManagerTest() throws SQLException {
+		super();
+	}
+	
+	@Autowired
+	private InitializerService iniz;
+	
+	@Autowired
+	private ReportService rs;
+	
+	@Autowired
+	private ReportDefinitionService rds;
+	
+	@Autowired
+	@Qualifier("conceptService")
+	private ConceptService cs;
+	
+	@Autowired
+	private LabReportManager manager;
+	
+	@Override
+	public void executeDataSet(IDataSet dataset) {
+		try {
+			Connection connection = getConnection();
+			IDatabaseConnection dbUnitConn = setupDatabaseConnection(connection);
+			DatabaseOperation.REFRESH.execute(dbUnitConn, dataset);
+		}
+		catch (Exception e) {
+			throw new DatabaseUnitRuntimeException(e);
+		}
+	}
+	
+	private IDatabaseConnection setupDatabaseConnection(Connection connection) throws DatabaseUnitException {
+		IDatabaseConnection dbUnitConn = new DatabaseConnection(connection);
+		
+		DatabaseConfig config = dbUnitConn.getConfig();
+		config.setProperty(DatabaseConfig.PROPERTY_DATATYPE_FACTORY, new MySqlDataTypeFactory());
+		
+		return dbUnitConn;
+	}
+	
+	@Before
+	public void setUp() throws Exception {
+		updateDatabase("org/openmrs/module/commonreports/liquibase/test-liquibase.xml");
+		executeDataSet("org/openmrs/module/reporting/include/ReportTestDataset-openmrs-2.0.xml");
+		executeDataSet("org/openmrs/module/commonreports/include/labTestDataset.xml");
+		
+		String path = getClass().getClassLoader().getResource("testAppDataDir").getPath() + File.separator;
+		System.setProperty("OPENMRS_APPLICATION_DATA_DIRECTORY", path);
+		
+		for (Loader loader : iniz.getLoaders()) {
+			if (loader.getDomainName().equals(Domain.JSON_KEY_VALUES.getName())) {
+				loader.load();
+			}
+		}
+	}
+	
+	@Test
+	public void setupReport_shouldCreateExcelTemplateDesign() throws Exception {
+		// setup
+		
+		// replay
+		ReportManagerUtil.setupReport(manager);
+		
+		// verif
+		ReportDesign design = rs.getReportDesignByUuid("988d3408-9aac-4b6f-9cab-77c3f02beef5");
+		
+		Assert.assertEquals(1, design.getResources().size());
+		
+		ReportDefinition def = design.getReportDefinition();
+		Assert.assertEquals("14f9b6a6-e176-467c-83ee-6281e743834d", def.getUuid());
+	}
+	
+	@Test
+	public void testReport() throws Exception {
+		// setup
+		EvaluationContext context = new EvaluationContext();
+		context.addParameterValue("startDate", DateUtil.parseDate("2021-06-01", "yyyy-MM-dd"));
+		context.addParameterValue("endDate", DateUtil.parseDate("2021-07-01", "yyyy-MM-dd"));
+		
+		// replay
+		ReportDefinition rd = manager.constructReportDefinition();
+		ReportData data = rds.evaluate(rd, context);
+		
+		// verify
+		for (Iterator<DataSetRow> itr = data.getDataSets().get(rd.getName()).iterator(); itr.hasNext();) {
+			DataSetRow row = itr.next();
+			assertEquals(ONE, row.getColumnValue("sputum"));
```

Comment:
It appears this line fails when running the tests:
```
[ERROR] Failures: 
[ERROR]   LabReportManagerTest.testReport:135 expected:<1> but was:<0>
```

Reply:
From various trials, it appears that the tests are failing only on my machine.
I'll investigate this further later but we can merge this now.